### PR TITLE
Support klib in binary compatibility validator. Upgrade gradle and plugins.

### PR DESCRIPTION
**Pull request by IlyaGulya**
## Upgrades:
Gradle `8.0.2` -> `8.4`
Kotlin `1.9.20` -> `1.9.23`
Gradle Intellij Plugin `1.15.0` -> `1.17.3`
AGP `8.0.2` -> `8.2.0`

## Improvements
- Added support for `klib` in binary compatibility validator

### CODE REVIEW

File: src/main/java/com/arkivanov/gradle/SetupJsApp.kt [hunk 1]
```diff
@@ -2,11 +2,11 @@ package com.arkivanov.gradle
 
 import org.gradle.api.Project
 import org.gradle.kotlin.dsl.configure
-import org.jetbrains.kotlin.gradle.dsl.KotlinJsProjectExtension
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
 
 fun Project.setupJsApp() {
-    extensions.configure<KotlinJsProjectExtension> {
-        js(IR) {
+    extensions.configure<KotlinMultiplatformExtension> {
```
Comment (arkivanov) [inline src/main/java/com/arkivanov/gradle/SetupJsApp.kt]: Do we really need to do this change now?

File: build.gradle.kts [hunk 2]
```diff
@@ -27,10 +27,10 @@ repositories {
 }
 
 dependencies {
-    compileOnly("org.jetbrains.kotlin:kotlin-gradle-plugin:1.9.20")
-    compileOnly("com.android.tools.build:gradle:8.0.2")
-    compileOnly("org.jetbrains.intellij.plugins:gradle-intellij-plugin:1.15.0")
-    compileOnly("org.jetbrains.kotlinx:binary-compatibility-validator:0.13.2")
+    compileOnly("org.jetbrains.kotlin:kotlin-gradle-plugin:1.9.23")
+    compileOnly("com.android.tools.build:gradle:8.3.1")
```
Comment (arkivanov) [inline build.gradle.kts]: Do we really need to update AGP and Gradle now? I'm trying to keep versions consistent across all projects that use this plugin.

File: build.gradle.kts [hunk 3]
```diff
@@ -27,10 +27,10 @@ repositories {
...
+    compileOnly("org.jetbrains.intellij.plugins:gradle-intellij-plugin:1.17.3")
+    compileOnly("org.jetbrains.kotlinx:binary-compatibility-validator:0.15.0-Beta.2")
```
Comment (arkivanov) [inline build.gradle.kts:33]: I suggest to wait for version 0.15.0. WDYT?

File: build.gradle.kts (hunk 2 above)
Comment (IlyaGulya) [reply build.gradle.kts]: Ok 🙂

File: src/main/java/com/arkivanov/gradle/SetupJsApp.kt (hunk 1 above)
Comment (IlyaGulya) [reply src/main/java/com/arkivanov/gradle/SetupJsApp.kt]: I'm not sure. Usually I'm trying to remove usages of deprecated APIs whenever possible. I can revert this change, for sure.

File: build.gradle.kts (hunk 3 above)
Comment (IlyaGulya) [reply build.gradle.kts:33]: As you wish 🙂 
I will make WIP PRs in all your libraries and mark this one as WIP as well.

BTW, Redwood already adopted it recently, that's why I've decided that it seems to be about stable 🙂  

https://github.com/cashapp/redwood/commit/aac293ce15d713c6dd85abbf3abd058f49eff085#diff-a5040955ccae69f8272686096ca4076bf8c69114111afecfd09c50503df78e4f

Comment (arkivanov) [reply build.gradle.kts:33]: Ok, let's keep this version for now. Anyway, it's not production.

File: build.gradle.kts (hunk 2 above)
Comment (arkivanov) [reply build.gradle.kts]: I'm using IntelliJ IDEA, I guess we could update Gradle to 8.4 and AGP to 8.2.0.

File: src/main/java/com/arkivanov/gradle/SetupJsApp.kt (hunk 1 above)
Comment (IlyaGulya) [reply src/main/java/com/arkivanov/gradle/SetupJsApp.kt]: Reverted this one

File: build.gradle.kts (hunk 2 above)
Comment (IlyaGulya) [reply build.gradle.kts]: Done

**Review by arkivanov — APPROVED** [review]

**Comment by arkivanov** [comment]
Thank you!

**Comment by arkivanov** [comment]
The new version is `b42078d459`.

**Pull request merged**